volatility3/framework/symbols/windows/extensions/mft.py

```diff
@@ -2,23 +2,152 @@
 # which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
 #
 
-from typing import Optional
+import logging
+from typing import Dict, Iterator, List, Optional, Tuple
 
-from volatility3.framework import objects, constants, exceptions
+from volatility3.framework import constants, exceptions, interfaces, objects
+
+vollog = logging.getLogger(__name__)
 
 
 class MFTEntry(objects.StructType):
     """This represents the base MFT Record"""
 
-    def get_signature(self) -> str:
+    def __init__(
+        self,
+        context: interfaces.context.ContextInterface,
+        type_name: str,
+        object_info: interfaces.objects.ObjectInformation,
+        size: int,
+        members: Dict[str, Tuple[int, interfaces.objects.Template]],
+        **kwargs,
+    ) -> None:
+        super().__init__(context, type_name, object_info, size, members)
+
+        self._symbol_table_name = kwargs.get("symbol_table_name")
```

Comment:
Assuming the needed types are in the same table as this type, you can get this information from self.vol.type_name (there's helper functions to get the symbol table part of it, I think)?  I'd be much happier with that than requiring it as an extra argument that can cause exceptions if it hasn't been initialized...

Reply:
Awesome thanks! I'll figure out how to get it from there instead.